packages/smooth_app/lib/widgets/smooth_product_carousel.dart

```diff
@@ -271,34 +272,17 @@ class _SearchCardTagLine extends StatelessWidget {
             if (preferences.isFirstScan) {
               return const _SearchCardTagLineDefaultText();
             }
-
-            return FutureBuilder<TagLineItem?>(
-              future: fetchTagLine(Platform.localeName),
-              builder:
-                  (BuildContext context, AsyncSnapshot<TagLineItem?> data) {
-                if (data.data == null) {
-                  return const _SearchCardTagLineDefaultText();
-                } else {
-                  return InkWell(
-                    borderRadius: ANGULAR_BORDER_RADIUS,
-                    onTap: data.data!.hasLink
-                        ? () async {
-                            if (await canLaunchUrlString(data.data!.url)) {
-                              await launchUrl(
-                                Uri.parse(data.data!.url),
-                                // forms.gle links are not handled by the WebView
-                                mode: LaunchMode.externalApplication,
-                              );
-                            }
-                          }
-                        : null,
-                    child: AutoSizeText(
-                      data.data!.message,
-                      style: TextStyle(
-                        color: Theme.of(context).colorScheme.primary,
-                      ),
-                    ),
+            return FutureBuilder<dynamic>(
+              future: _fetchData(),
+              builder: (BuildContext context, AsyncSnapshot<dynamic> data) {
+                if (data.data != null && data.data is! TagLineItem) {
+                  return const _SearchCardTagLineDeprecatedAppText();
+                } else if (data is TagLineItem) {
+                  return _SearchCardTagLineText(
```

Comment:
No strong added value to move that code into a separate widget: please think about the poor developer that will review your code ;)

Reply:
@g123k review the comment from the code you sent me